drools-drl-quarkus-extension/drools-drl-quarkus-deployment/src/main/java/org/drools/drl/quarkus/deployment/DroolsAssetsProcessor.java

```diff
@@ -0,0 +1,97 @@
+import org.kie.api.io.Resource;
+
+import static org.drools.drl.quarkus.deployment.DroolsQuarkusResourceUtils.HOT_RELOAD_SUPPORT_PATH;
+import static org.drools.drl.quarkus.deployment.DroolsQuarkusResourceUtils.compileGeneratedSources;
+import static org.drools.drl.quarkus.deployment.DroolsQuarkusResourceUtils.createDroolsBuildContext;
+import static org.drools.drl.quarkus.deployment.DroolsQuarkusResourceUtils.dumpFilesToDisk;
+import static org.drools.drl.quarkus.deployment.DroolsQuarkusResourceUtils.getHotReloadSupportSource;
+import static org.drools.drl.quarkus.deployment.DroolsQuarkusResourceUtils.registerResources;
+import static org.drools.model.project.codegen.RuleCodegen.ofResources;
+
+public class DroolsAssetsProcessor {
+
+    @Inject
+    ArchiveRootBuildItem root;
+    @Inject
+    LiveReloadBuildItem liveReload;
+    @Inject
+    CurateOutcomeBuildItem curateOutcomeBuildItem;
+    @Inject
+    CombinedIndexBuildItem combinedIndexBuildItem;
+    @Inject
+    OutputTargetBuildItem outputTargetBuildItem;
+
+    private static final String FEATURE = "drools-drl-quarkus-extension";
+
+    @BuildStep
+    public FeatureBuildItem feature() {
+        return new FeatureBuildItem(FEATURE);
+    }
+
+    @BuildStep
+    public void generateSources( BuildProducer<GeneratedBeanBuildItem> generatedBeans,
+                                 BuildProducer<NativeImageResourceBuildItem> resource,
+                                 BuildProducer<AdditionalStaticResourceBuildItem> staticResProducer,
+                                 BuildProducer<GeneratedResourceBuildItem> genResBI ) {
+        DroolsModelBuildContext context =
+                createDroolsBuildContext(outputTargetBuildItem.getOutputDirectory(), root.getPaths(), combinedIndexBuildItem.getIndex());
+
+        Collection<Resource> resources = ResourceCollector.fromPaths(context.getAppPaths().getPaths());
+
+        Collection<GeneratedFile> generatedFiles = ofResources(context, resources).generate();
+
+        // The HotReloadSupportClass has to be generated only during the first model generation
+        // During actual hot reloads it will be regenerated by the compilation providers in order to retrigger this build step
+        if (!liveReload.isLiveReload()) {
+            generatedFiles.add(new GeneratedFile(GeneratedFileType.SOURCE, HOT_RELOAD_SUPPORT_PATH + ".java", getHotReloadSupportSource()));
+        }
+
+        // dump files to disk
+        dumpFilesToDisk(context.getAppPaths(), generatedFiles);
+
+        // build Java source code and register the generated beans
+        Collection<ResolvedDependency> dependencies = curateOutcomeBuildItem.getApplicationModel().getRuntimeDependencies();
+
+        Collection<GeneratedBeanBuildItem> generatedBeanBuildItems =
+                compileGeneratedSources(context, dependencies, generatedFiles, liveReload.isLiveReload());
+        generatedBeanBuildItems.forEach(generatedBeans::produce);
```

Comment:
This comment should be moved to the other line
```suggestion
        Collection<ResolvedDependency> dependencies = curateOutcomeBuildItem.getApplicationModel().getRuntimeDependencies();

        // build Java source code and register the generated beans
        Collection<GeneratedBeanBuildItem> generatedBeanBuildItems =
                compileGeneratedSources(context, dependencies, generatedFiles, liveReload.isLiveReload());
        generatedBeanBuildItems.forEach(generatedBeans::produce);
```

Reply:
Fixed.